blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java

```diff
@@ -234,7 +234,19 @@ public HttpResponse submitInputStep(StaplerRequest request) {
             }
 
             //XXX: execution.doProceed(request) expects submitted form, otherwise we could have simply used it
-            execution.preSubmissionCheck();
+
+            try {
+                execution.preSubmissionCheck();
+            } catch (Failure f) {
+                return new HttpResponse() {
+                    @Override
+                    public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object node) throws IOException, ServletException {
+                        rsp.setStatus(400);
+                        rsp.setContentType("application/json;charset=UTF-8");
+                        rsp.getWriter().print(JsonConverter.toJson(new ErrorMessage(400, f.getMessage())));
```

Comment:
Would throwing a `ServiceException` here instead of returning `new HttpResponse()` be a simpler way to achieve the same effect? @vivek

Reply:
Yes, maybe worth trying ‘BadRequestException’.